#! /bin/sh /usr/share/dpatch/dpatch-run
## 100-kolab-imapd.dpatch by Steffen Joeris <steffen.joeris@skolelinux.de>
##
## DP: Kolab upstream's imapd.patch adapted

@DPATCH@
--- a/et/com_err.c
+++ b/et/com_err.c
@@ -51,7 +51,7 @@
 #include <string.h>
 #include "mit-sipb-copyright.h"
 
-#if defined(HAVE_STDARG_H) || defined(_WINDOWS)
+#if defined(HAVE_STDARG_H) || defined(__STDC__) || defined(_WINDOWS)
 #include <stdarg.h>
 #else
 #include <varargs.h>
--- a/lib/cyrusdb_skiplist.c
+++ b/lib/cyrusdb_skiplist.c
@@ -71,6 +71,11 @@
 
 #define PROB (0.5)
 
+#ifdef __FreeBSD__
+/* #define fdatasync(fd) fsync(fd) */
+#define O_DSYNC 0
+#endif
+
 /* 
  *
  * disk format; all numbers in network byte order
--- a/lib/prot.h
+++ b/lib/prot.h
@@ -49,6 +49,7 @@
 #include <time.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/time.h>
 
 #include <sasl/sasl.h>
 
--- a/imap/mboxname.c
+++ b/imap/mboxname.c
@@ -128,7 +128,11 @@ static int mboxname_tointernal(struct na
 		sprintf(result, "%s!", cp);
 	    }
 	}
+#ifdef ATVDOM /* allow '@' being a regular character in mboxname even when using virtual domains */
+	else if ((cp = strrchr(name, '@'))) {
+#else
 	if ((cp = strrchr(name, '@'))) {
+#endif /* ATVDOM */
 	    /* mailbox specified as mbox@domain */
 	    namelen = cp - name;
 
